cmd/podman/common/create.go

```diff
@@ -897,19 +897,23 @@ func DefineCreateFlags(cmd *cobra.Command, cf *entities.ContainerCreateOptions,
 		_ = cmd.RegisterFlagCompletionFunc(memorySwappinessFlagName, completion.AutocompleteNone)
 	}
 	// anyone can use these
-	cpusFlagName := "cpus"
-	createFlags.Float64Var(
-		&cf.CPUS,
-		cpusFlagName, 0,
-		"Number of CPUs. The default is 0.000 which means no limit",
-	)
-	_ = cmd.RegisterFlagCompletionFunc(cpusFlagName, completion.AutocompleteNone)
-
-	cpusetCpusFlagName := "cpuset-cpus"
-	createFlags.StringVar(
-		&cf.CPUSetCPUs,
-		cpusetCpusFlagName, "",
-		"CPUs in which to allow execution (0-3, 0,1)",
-	)
-	_ = cmd.RegisterFlagCompletionFunc(cpusetCpusFlagName, completion.AutocompleteNone)
+	// CHECK IF ALREADY DEFINED! this function can be used twice for some commands
+	if _, err := createFlags.GetFloat64("cpus"); err != nil {
```

Comment:
This looks very bad, how can this happen? We should not duplicate flags.

Reply:
Let me check on this, that might be an incorrect assumption.

Reply:
oh yeah @Luap99 if you look a https://github.com/containers/podman/blob/80619a3b1936a769c92f7319b589fd35696c4254/cmd/podman/pods/clone.go#L47-L55 we need both of these since 1) eventually I will be adding pod specific options and 2) I will be adding container specific options. However, there are some things only to be exposed at the pod level (resource limits for example) meaning if we go to define these options again in the `PerContainerOptions` call, cobra will error out.

Reply:
We still should not define the same flag twice, if this is the case then the code flow is wrong and IMO becomes unmaintainable. How can this happen, on your second call you set it to a different command which should not have these flags defined?

Reply:
This means that you call this function twice with the same cmd. This does not seem to be the case since your second call uses an empty cmd. Therefore I am unable to understand why this is required.

Reply:
I am going to create a function in cmd/common that sets defaults.

Reply:
good idea. I will do it that way and use the default function everywhere that `DefineCreateFlags` is used.